src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas3/ReportService.kt

```diff
@@ -55,4 +54,15 @@ class ReportService(
         WorkbookFactory.create(true)
       }
   }
+
+  private fun splitAndRetrievePersonInfo(crns: Set<String>): Map<String, PersonSummaryInfoResult> {
+    val deliusUsername = userService.getUserForRequest().deliusUsername
+
+    val crnMap = ListUtils.partition(crns.toList(), numberOfCrn)
+      .stream().map { crns ->
+        offenderService.getOffenderSummariesByCrns(crns.toSet(), deliusUsername).associateBy { it.crn }
+      }.collect(Collectors.toList())
+
+    return crnMap.flatMap { it.toList() }.toMap()
+  }
```

Comment:
as per previous comment, moving this logic into ApDeliusContextApiOffenderDetailsDataSource would remove duplication between these two services

Reply:
Since the fix is isolated into single place now would give confident of the fix before we can move into common place if needed.